### ISSUE TITLE
Split second graphics issue.

### ISSUE THREAD

**Issue by 71knight** [hrydgard/ppsspp#15785]
### Game or games this happens in

Split / second

### What area of the game

Port bridge racetrack.

### What happens

When driving under the building, the sun is visible through the roof.

![splitsecond](https://user-images.githubusercontent.com/62399979/182765725-a0cab863-700c-4d21-903a-621318b69195.jpg)

### What should happen

The sun shouldn't be visible through the roof.

### Platform

Android

### Mobile phone model or graphics card

OnePlus 7t with adreno 640.

### PPSSPP version affected

Ver 1.13.1-130

### Last working version

N/A

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try changing graphics settings to determine if one causes the glitch (especially speed hacks or enhancements/replacements.)
- [X] Include logs or screenshots of issue.

**Comment by Gamemulatorer** [comment]
I can also reproduce this on my redmi note 9 mali gpu using ppsspp v1.13.1-153

**Comment by Gamemulatorer** [comment]
Monster Hunter 3rd HD also have this graphics issue.
Taking reward the background should be black and not the image of my savestate.
![Screenshot_2022-08-04-18-27-55-552_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/182826112-3b28df6c-7714-4170-a736-b6c46a292ebe.jpg)

**Comment by Panderner** [comment]
@Gamemulatorer that correct background is not right

This how it looks like on PS3:
![Screenshot (12)](https://user-images.githubusercontent.com/37503397/182986916-42f51819-9108-42c7-b322-ffbd7f5b7e06.png)

**Comment by Gamemulatorer** [comment]
> @Gamemulatorer that correct background is not right
> 
> This how it looks like on PS3: ![Screenshot (12)](https://user-images.githubusercontent.com/37503397/182986916-42f51819-9108-42c7-b322-ffbd7f5b7e06.png)

How about on real psp?

**Comment by Panderner** [comment]
> How about on real psp?

The HD version does not work for real PSP consoles Only works for PS3 and This emulator. Real PSP only works for Vanilla version of this game.

**Comment by Gamemulatorer** [comment]
Can you try in training mode after completing the quest take the rewards in screenshot.

**Comment by Gamemulatorer** [comment]
> > How about on real psp?
> 
> The HD version does not work for real PSP consoles Only works for PS3 and This emulator. Real PSP only works for Vanilla version of this game.

This is software rendering ppsspp
![Screenshot_2022-08-07-09-16-25-273_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/183278118-f4908eed-9aeb-4f35-b7d4-dc10d7e5b931.jpg)

**Comment by Panderner** [comment]
![Screenshot (13)](https://user-images.githubusercontent.com/37503397/183281229-0d6b8b05-5996-4928-a4a7-b6f8cda29336.png)
Here's a screenshot for vanilla MHP3rd using software renderer

**Comment by hrydgard** [comment]
Wasn't this issue about Split/Second? If MHP3rd has issues, make a new issue and move the images please :)

**Comment by Gamemulatorer** [comment]
ppsspp v1.13.1-216
![Screenshot_2022-08-07-17-20-45-291_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/183284123-2c5a4dc9-95da-4fab-8634-cd3b75f5e0fa.jpg)
[ULUS10513.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/9276521/ULUS10513.ppdmp.zip)

**Comment by Gamemulatorer** [comment]
> ![Screenshot (13)](https://user-images.githubusercontent.com/37503397/183281229-0d6b8b05-5996-4928-a4a7-b6f8cda29336.png) Here's a screenshot for vanilla MHP3rd using software renderer

It's ok now Panderner 👍
![Screenshot_2022-08-07-21-13-19-762_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/183293245-beb7e066-023c-42ae-89c5-5a1f31de72f6.jpg)
Thanks for reply/feedback!

**Comment by unknownbrackets** [comment]
> ppsspp v1.13.1-216 ![Screenshot_2022-08-07-17-20-45-291_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/183284123-2c5a4dc9-95da-4fab-8634-cd3b75f5e0fa.jpg) [ULUS10513.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/9276521/ULUS10513.ppdmp.zip)

For me, software and Vulkan look somewhat similar, like this:
![Sun visible through building](https://user-images.githubusercontent.com/191233/183297552-0468f2b5-6766-4e2d-bf07-1e097f5a6995.png)

I'm not sure if that's correct, but it's the same for both.  If you take a frame dump while using software rendering, does it not happen (using that dump) even in hardware?  That would be interesting.

-[Unknown]

**Comment by 71knight** [comment]
I believe this problem is solved by using hardware read backs, at least in aethersx2 emulator it is.  When I disable hardware readbacks in aethersx2, the sun shines through every solid object.  When I enable hardware readbacks, the sun is occluded by solid objects passing in front of it.  I guess PPSSPP doesn't have that feature implemented yet?

**Comment by Gamemulatorer** [comment]
This issue already exist since v1.13.0
![Screenshot_2022-08-08-19-22-41-210_org ppsspp ppssppgold](https://user-images.githubusercontent.com/37603562/183407184-9a3ad555-1715-4ce7-a7a9-7aef8905aca5.jpg)
PPSSPP GOLD v1.13.1 (bought on playstore)

**Comment by hrydgard** [comment]
Yeah, I haven't confirmed exactly how the game determines whether the sun should be visible or not, but the most likely is that it's using the CPU to read the depth or color buffer. This will work in the software renderer, but not in the hardware renderer without forcing on some kind of readbacks. 

Some direct investigation is needed.

**Comment by 71knight** [comment]
I just wanted to add an observation that I made.  When the sun is at normal  brightness intensity, it doesn't shine through solid objects.  But when the sun increases its intensity and cast a lens flare, then the sun will shine through solid objects such as a building or roof.  Best seen on the docks stage near the big boat thats stationary.  Hopefully this information will help.  Thank you for all the hard work the team put into this awesome emulator.

**Comment by Gamemulatorer** [comment]
> Unfortunately that'll break the HUD :/
Yeah forget to say that 😅

> Does the sun effect work at all without it?
No, need BlueToAlpha enable to make the sun flares work. maybe this issue is related to this now https://github.com/hrydgard/ppsspp/issues/11100

**Comment by Gamemulatorer** [comment]
> > ppsspp v1.13.1-216 ![Screenshot_2022-08-07-17-20-45-291_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/183284123-2c5a4dc9-95da-4fab-8634-cd3b75f5e0fa.jpg) [ULUS10513.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/9276521/ULUS10513.ppdmp.zip)
> 
> For me, software and Vulkan look somewhat similar, like this: ![Sun visible through building](https://user-images.githubusercontent.com/191233/183297552-0468f2b5-6766-4e2d-bf07-1e097f5a6995.png)
> 
> I'm not sure if that's correct, but it's the same for both. If you take a frame dump while using software rendering, does it not happen (using that dump) even in hardware? That would be interesting.
> 
> -[Unknown]

Loading the ge dump both hardware and software this issue can still reproduce.
But loading the game normally and use software rendering the graphics is correct.
![Screenshot_2022-08-09-00-55-44-056_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/183483477-4e9c2548-eb3a-463c-bd28-71ed45d0443a.jpg)

Opengl show some graphics warning that I experience recently on my android 5 phone.
![Screenshot_2022-08-09-00-56-13-448_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/183483605-22462308-488a-48e4-9d8c-307990726825.jpg)

**Comment by hrydgard** [comment]
I'm gonna fix that shader warning.

So yeah, bluetoalpha fixed so that the sun flare is visible, but it does not resolve the issue that it's shining through. I wouldn't say then that bluetoalpha "broke" it, rather that there is an additional separate issue to fix.

Multiple games have problems with occluded lens flares, they're usually done by some hacky readback tricks since there's no good "official" way to do them on the PSP. Sometime in the near future I'm gonna go through these and see what the games are actually doing (I only know for sure what Burnout Dominator is doing, I have a suspicion about Wipeout), and what we can do to make it work without slowing things down too much.

**Comment by Gamemulatorer** [comment]
I also notice that minimizing ppsspp had this weird issue on Split/Second.

Suns Flares Disable (Disable Slower Effects ON)

https://user-images.githubusercontent.com/37603562/183535045-3c465319-9d5b-48e6-ab6f-ac2e56c1cea2.mp4

Sun Flares Enable

https://user-images.githubusercontent.com/37603562/183535087-7ee9c6f3-c9b3-42bc-bb9f-a3143ff76d4a.mp4

**Comment by hrydgard** [comment]
That is a bit weird that an old screenshot shows up, but doesn't seem to hinder playability luckily.
